File: skellig-test-step-processing/src/main/java/org/skellig/teststep/processing/converter/TestStepResultToStringConverter.java
package org.skellig.teststep.processing.converter;

import java.nio.charset.StandardCharsets;

public class TestStepResultToStringConverter implements TestStepResultConverter {

    @Override
    public Object convert(String convertFunction, Object result) {
        if (result instanceof byte[]) {
            return new String((byte[]) result, StandardCharsets.UTF_8);
        } else {
            return String.valueOf(result);
        }
    }

    @Override
    public String getConvertFunctionName() {
        return "string";
    }
}

File: skellig-test-step-processing/src/main/kotlin/org/skellig/teststep/processing/converter/TestStepResultToStringConverter.kt
package org.skellig.teststep.processing.converter

import java.nio.charset.StandardCharsets

class TestStepResultToStringConverter : TestStepResultConverter {

     override fun convert(convertFunction: String, result: Any?): Any? {
        return if (result is ByteArray) {
            String(result, StandardCharsets.UTF_8)
        } else {
            result.toString()
        }
    }

    override fun getConvertFunctionName(): String {
        return "string"
    }
}